cli/rbd/rbd.py

```diff
@@ -0,0 +1,315 @@
+from copy import deepcopy
+
+from cli import Cli
+from cli.utilities.utils import build_cmd_from_args
+
+from .device import Device
+from .mirror.mirror import Mirror
+from .pool import Pool
+from .snap import Snap
+
+
+class Rbd(Cli):
+    def __init__(self, nodes, base_cmd=""):
+        super(Rbd, self).__init__(nodes)
+        self.base_cmd = f"{base_cmd}rbd"
+        self.pool = Pool(nodes, self.base_cmd)
+        self.mirror = Mirror(nodes, self.base_cmd)
+        self.device = Device(nodes, self.base_cmd)
+        self.snap = Snap(nodes, self.base_cmd)
+
+    def create(self, **kw):
+        """
+        Creates a block device image.
+        Args:
+        kw(dict): Key/value pairs that needs to be provided to the installer
+            Example::
+            Supported keys:
+                image-spec(str) : <pool_name>/<image_name>
+                pool(str) : name of the pool into which image should be stored
+                size(str) : size of image
+                image(str): name of image to be created
+                See rbd help create for more supported keys
+        """
+        kw_copy = deepcopy(kw)
+        image_spec = kw_copy.pop("image-spec", "")
+        cmd = f"{self.base_cmd} create {image_spec} {build_cmd_from_args(**kw_copy)}"
+
+        return self.execute_as_sudo(cmd=cmd)
+
+    def ls(self, **kw):
+        """
+        Lists block device images within pool.
+        Args:
+        kw(dict): Key/value pairs that needs to be provided to the installer
+            Example::
+            Supported keys:
+                pool(str)  : name of the pool(optional)
+                pool-spec(str) : <pool_name>[/<namespace>]
+                See rbd help ls for more supported keys
+        """
+        kw_copy = deepcopy(kw)
+        pool_spec = kw_copy.pop("pool-spec", "")
+        cmd = f"{self.base_cmd} ls {pool_spec} {build_cmd_from_args(**kw_copy)}"
+
+        return self.execute(cmd=cmd)
+
+    def list_(self, **kw):
+        """
+        Lists block device images.
+        Args:
+        kw(dict): Key/value pairs that needs to be provided to the installer
+            Example::
+            Supported keys:
+                pool(str)  : name of the pool(optional)
+                pool-spec(str) : <pool_name>[/<namespace>]
+                See rbd help list for more supported keys
+        """
+        kw_copy = deepcopy(kw)
+        pool_spec = kw_copy.pop("pool-spec", "")
+        cmd = f"{self.base_cmd} ls {pool_spec} {build_cmd_from_args(**kw_copy)}"
```

Comment:
Yes It was adopted from Vasi's PR looks like its just another wrapper to use list_ instead of ls if required.

Reply:
[root@ceph-rbd1-k920-bz-g2gxd5-node2 ~]# rbd help list
usage: rbd list [--long] [--pool <pool>] [--namespace <namespace>] 
                [--format <format>] [--pretty-format] 
                <pool-spec> 
                
I will modify the list_ method to call rbd list instead of rbd ls.

Reply:
Changes Done.